Source/engine/render/automap_render.cpp

```diff
@@ -24,32 +26,32 @@ template <DirectionX DirX, DirectionY DirY>
 void DrawMapLine(const Surface &out, Point from, int height, std::uint8_t colorIndex)
 {
 	while (height-- > 0) {
-		out.SetPixel({ from.x, from.y + 1 }, 0);
-		out.SetPixel(from, colorIndex);
+		(!AutomapTransparent) ? out.SetPixel({ from.x, from.y + 1 }, 0) : SetHalfTransparentPixel(out, { from.x, from.y + 1 }, 0);

```

Comment:
Please introduce and use a function along these lines:

```c++
void SetMapPixel(const Surface &out, Point point, uint8_t color)
{
	if (AutomapTransparent) {
		out.SetPixel(point, color);
	} else {
		SetHalfTransparentPixel(out, point, color);
	}
}
```

Reply:
Such a simple solution, I wish I would have thought of that! The way I did this was annoying and painful to do :(